### PR TITLE
Crm 17120

### PR DESCRIPTION
**Pull request by eileenmcnaughton**

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
Note the PR is mostly tests & comments

**Comment by colemanw** [comment]
I think stuff like this generally goes into master rather than a point release.

**Comment by eileenmcnaughton** [comment]
Yeah - it looks like a whole lot more change than it is because I'm adding tests & fixing comments. 

But if I go straight to master I don't need to be as careful about adding tests :-)

File: CRM/Case/BAO/Query.php [hunk 1]
```diff
@@ -235,7 +229,6 @@ public static function where(&$query) {
         if ($query->_mode == CRM_Contact_BAO_Query::MODE_CONTACTS) {
           $query->_useDistinct = TRUE;
         }
-        $grouping = $query->_params[$id][3];
```
Comment (colemanw) [inline CRM/Case/BAO/Query.php]: What's this change doing? Seems unrelated to the stated issue.

Comment (eileenmcnaughton) [inline CRM/Case/BAO/Query.php]: It's an unused variable. I think I removed the same line in master in a sibling search class - or I might not have committed that - but I think it's a child of copy & paste

**Comment by colemanw** [comment]
Ok well if you're confident that tests are covering it then I guess it's safe for 4.6.

**Comment by eileenmcnaughton** [comment]
unrelated failure

**Comment by eileenmcnaughton** [comment]
Yes, these are heavily tested. I'll look at which of the remaining ones are being called a lot & decide how to deal with the rest (or perhaps whether to only do a select few of them that have the most obvious performance impact - but I WOULD like to completely remove that function).

The trade-off between master vs 4.6 is that if I do them in master the time goes into wrangling patches, if I do them in 4.6 then the time goes into writing unit tests. I can't realistically do both.

**Comment by colemanw** [comment]
You make a good case. I'd merge it but you already did :)
